osquery/filesystem/linux/proc.cpp

```diff
@@ -396,4 +399,34 @@ Status procReadDescriptor(const std::string& process,
   }
 }
 
+Expected<std::uint64_t, ProcError> getProcRSS(const std::string& process) {
+  using ProcExpected = Expected<std::uint64_t, ProcError>;
+
+  std::string statm_content;
+  auto status = osquery::readFile(kLinuxProcPath + "/" + process + "/statm",
```

Comment:
any idea what the performance hit here will be?

Reply:
Should be almost invisible. We are talking about reading maybe 50 bytes which are always in memory, since it's a pseudofile.